internal/provisioner/cluster.go

```diff
@@ -234,6 +258,38 @@ func DeleteCluster(clusterID, s3StateStore string, logger log.FieldLogger) error
 	return nil
 }
 
+// waitForKubernetesReadyness will poll a given kubernetes cluster at a regular
+// interval for it to become ready. If the cluster fails to become ready before
+// the provided timeout then an error will be returned.
+func waitForKubernetesReadyness(dns string, kops *kops.Cmd, timeout time.Duration) error {
+	ready := make(chan bool, 1)
+	defer close(ready)
+
+	go func() {
+		for {
+			err := kops.ValidateCluster(dns, true)
+			if err == nil {
+				break
+			}
+			time.Sleep(5 * time.Second)
+		}
+
+		ready <- true
```

Comment:
suggest `defer close(ready)` at the top of the function, and changing `break` to `return`.